common/Resources/ti.internal/extensions/node/internal/errors.js

```diff
@@ -0,0 +1,205 @@
+function E(sym, val, def, ...otherClasses) {
+	// Special case for SystemError that formats the error message differently
+	// The SystemErrors only have SystemError as their base classes.
+	messages.set(sym, val);
+
+	if (def === SystemError) {
+		throw new Error('Node compatible SystemError not yet implemented.');
+	} else {
+		def = makeNodeErrorWithCode(def, sym);
+	}
+
+	if (otherClasses.length !== 0) {
+		otherClasses.forEach((clazz) => {
+			def[clazz.name] = makeNodeErrorWithCode(clazz, sym);
+		});
+	}
+	codes[sym] = def;
+}
+
+function makeNodeErrorWithCode(Base, key) {
+	return class NodeError extends Base {
+		constructor(...args) {
+			super();
+			const message = getMessage(key, args, this);
+			Object.defineProperty(this, 'message', {
+				value: message,
+				enumerable: false,
+				writable: true,
+				configurable: true
+			});
+			addCodeToName(this, super.name, key);
+		}
+
+		get code() {
+			return key;
+		}
+
+		set code(value) {
+			Object.defineProperty(this, 'code', {
+				configurable: true,
+				enumerable: true,
+				value,
+				writable: true
+			});
+		}
+
+		toString() {
+			return `${this.name} [${key}]: ${this.message}`;
+		}
+	};
+}
+
+function getMessage(key, args, self) {
+	const msg = messages.get(key);
+
+	/*
+	// @fixme rollup cannot handle lazy loaded modules, maybe move to webpack?
+	if (assert === undefined) {
+		assert = require('./internal/assert');
+	}
+	*/
+
+	if (typeof msg === 'function') {
+		assert(
+			msg.length <= args.length, // Default options do not count.
+			`Code: ${key}; The provided arguments length (${args.length}) does not ` +
+				`match the required ones (${msg.length}).`
+		);
+		return msg.apply(self, args);
+	}
+
+	const expectedLength = (msg.match(/%[dfijoOs]/g) || []).length;
+	assert(
+		expectedLength === args.length,
+		`Code: ${key}; The provided arguments length (${args.length}) does not ` +
+			`match the required ones (${expectedLength}).`
+	);
+	if (args.length === 0) {
+		return msg;
+	}
+
+	args.unshift(msg);
+	return format.apply(null, args);
+	// @fixme rollup cannot handle lazy loaded modules, maybe move to webpack?
+	// return lazyInternalUtilInspect().format.apply(null, args);
+}
+
+function addCodeToName(err, name, code) {
+	// Add the error code to the name to include it in the stack trace.
+	err.name = `${name} [${code}]`;
+	// Access the stack to generate the error message including the error code
+	// from the name.
+	// @fixme: This only works on V8/Android, iOS/JSC has a different Error structure.
+	// should we try to make errors behave the same across platforms?
+	err.stack;
+	// Reset the name to the actual name.
+	if (name === 'SystemError') {
+		Object.defineProperty(err, 'name', {
+			value: name,
+			enumerable: false,
+			writable: true,
+			configurable: true
+		});
+	} else {
+		delete err.name;
+	}
+}
+
+E('ERR_INTERNAL_ASSERTION', (message) => {
+	const suffix = 'This is caused by either a bug in Titanium ' +
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/internal/errors.js;
  Line: 131;
-->

- :warning: common/Resources/ti.internal/extensions/node/internal/errors.js line 131 – '+' should be placed at the beginning of the line. (operator-linebreak)